internal/resources/kv/kv.go

```diff
@@ -0,0 +1,105 @@
+package kv
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ydb-platform/ydb-go-genproto/draft/protos/Ydb_KeyValue"
+
+	"github.com/ydb-platform/terraform-provider-ydb/internal/helpers"
+)
+
+type MediaConfig struct {
+	Media string
+}
+
+type ChannelConfig struct {
+	Channel []*MediaConfig
+}
+
+type Resource struct {
+	Entity *helpers.YDBEntity
+
+	UseTLS           bool
+	FullPath         string
+	Endpoint         string
+	Database         string
+	Path             string
+	DatabaseEndpoint string
+	PartitionCount   int
+	StorageConfig    *ChannelConfig
+}
+
+func kvResourceSchemaToKvResource(d *schema.ResourceData) (*Resource, error) {
+	var entity *helpers.YDBEntity
+	var err error
+	if d.Id() != "" {
+		entity, err = helpers.ParseYDBEntityID(d.Id())
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse table entity: %w", err)
+		}
+	}
+
+	storageConfig, err := expandStorageConfig(d)
+	if err != nil {
+		return nil, errors.New("can't expandStorageConfig")
```

Comment:
return nil, fmt.Errorf("failed to expand storage config: %w", err)

Reply:
forgot, fixed